spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -1272,6 +1291,14 @@ private void logConsumerException(Throwable t) {
 				logger.warn("Consumer raised exception, processing can restart if the connection factory supports it. "
 						+ "Exception summary: " + t);
 			}
+			publishEvent("Consumer raised exception, attempting restart", false, t);
+		}
+
+		private void publishEvent(String reason, boolean fatal, Throwable t) {
```

Comment:
Why doesn't this method belong to the top-level `SimpleMessageListenerContainer` class ?